roles/openshift_master_facts/lookup_plugins/openshift_master_facts_default_predicates.py

```diff
@@ -0,0 +1,83 @@
+            return openshift_master_scheduler_predicates
+        elif 'scheduler_predicates' in variables['openshift']['master']:
+            return variables['openshift']['master']['scheduler_predicates']
+        else:
+            predicates = []
+
+            deployment_type = variables['openshift']['common']['deployment_type']
+            short_version = variables['openshift']['common']['short_version']
+
+            if deployment_type not in ['origin', 'openshift-enterprise']:
+                raise AnsibleError("Unknown deployment_type %s" %  deployment_type)
+
+            if deployment_type == 'origin':
+                if short_version not in ['1.1', '1.2', '1.3', '1.4']:
+                    raise AnsibleError("Unknown short_version %s" % short_version)
+            elif deployment_type == 'openshift_enterprise':
+                if short_version not in ['3.1', '3.2', '3.3', '3.4']:
+                    raise AnsibleError("Unknown short_version %s" % short_version)
+
+            if deployment_type == 'openshift-enterprise':
+                # convert short_version to origin short_version
+                short_version = re.sub('^3.', '1.', short_version)
+
+            display.vvv("short_version: %s" % short_version)
+
+            if short_version in ['1.1', '1.2']:
+                predicates.append({'name': 'PodFitsHostPorts'})
+                predicates.append({'name': 'PodFitsResources'})
+
+            # applies to all known versions
+            predicates.append({'name': 'NoDiskConflict'})
+
+            # only 1.1 didn't include NoVolumeZoneConflict
+            if short_version != '1.1':
+                predicates.append({'name': 'NoVolumeZoneConflict'})
+
+
+            if short_version in ['1.1', '1.2']:
+                predicates.append({'name': 'MatchNodeSelector'})
+                predicates.append({'name': 'Hostname'})
+
+            if short_version != '1.1':
+                predicates.append({'name': 'MaxEBSVolumeCount'})
+                predicates.append({'name': 'MaxGCEPDVolumeCount'})
+
+            if short_version not in ['1.1', '1.2']:
+                predicates.append({'name': 'GeneralPredicates'})
+                predicates.append({'name': 'PodToleratesNodeTaints'})
+                predicates.append({'name': 'CheckNodeMemoryPressure'})
+
+            if short_version not in ['1.1', '1.2', '1.3']:
+                predicates.append({'name': 'CheckNodeDiskPressure'})
+                predicates.append({'name': 'MatchInterPodAffinity'})
+
+            if regions_enabled:
+                region_predicate = {
+                        'name': 'Region',
+                        'argument': {
+                            'serviceAffinity': {
+                                'labels': ['region']
+                            }
+                        }
+                }
+                predicates.append(region_predicate)
+
+
+            return predicates
```

Comment:
What do you think about adding some unit tests for these modules? They look like great candidates for it, just need a little import magic for the test/ module and it should work fine.